app/src/main/java/org/p2p/wallet/infrastructure/network/provider/SeedPhraseProvider.kt

```diff
@@ -12,16 +12,36 @@ class SeedPhraseProvider(
 ) {
 
     val isAvailable: Boolean
-        get() = seedPhrase.isNotEmpty()
+        get() = getUserSeedPhrase().seedPhrase.isNotEmpty()
 
-    var seedPhrase: List<String> = emptyList()
+    fun getUserSeedPhrase(): UserSeedPhrase {
+        return UserSeedPhrase(
+            seedPhrase = seedPhrase,
+            provider = seedPhraseProvider
+        )
+    }
+
+    fun updateUserSeedPhrase(words: List<String>, provider: SeedPhraseProviderType) {
+        seedPhrase = words
+        seedPhraseProvider = provider
+    }
+
+    private var seedPhrase: List<String> = emptyList()
         get() = getSeedPhraseFromStorage()
         set(value) {
             field = value
             saveSeedPhraseToStorage(value)
             Timber.tag(TAG).i("updating user public key: $value")
         }
 
+    private var seedPhraseProvider: SeedPhraseProviderType? = null
+        get() = getSeedPhraseProviderFromStorage()
+        set(value) {
+            field = value
+            value?.let { saveSeedPhraseProviderToStorage(it) }
```

Comment:
if it's `null` you need to clear the storage?

Reply:
hmm, I think no